pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -122,18 +122,21 @@ public HelixBrokerStarter(String helixClusterName, String zkServer, Configuratio
     _helixAdmin = _helixManager.getClusterManagmentTool();
     addInstanceTagIfNeeded(helixClusterName, brokerId);
 
-
     // Register the service status handler
     ServiceStatus.setServiceStatusCallback(
         new ServiceStatus.IdealStateAndExternalViewMatchServiceStatusCallback(_helixAdmin, helixClusterName, brokerId));
 
     ClusterChangeMediator clusterChangeMediator = new ClusterChangeMediator(_helixExternalViewBasedRouting,
         _brokerServerBuilder.getBrokerMetrics());
-    _helixManager.addExternalViewChangeListener(clusterChangeMediator);
     _helixManager.addInstanceConfigChangeListener(clusterChangeMediator);
```

Comment:
InstanceConfig and LiveInstance change listener must also be registered to spectatorManager